src/redis_info_provider/info_poller.py

```diff
@@ -42,6 +45,15 @@ def __init__(self, grace=3):
         for shard in ShardPublisher.get_live_shards():
             self._add_shard(shard)
 
+    def set_wait(self, wait_on):
+        # type: (AsyncResult) -> None
+        self._wait_on = wait_on
+
+    def terminate(self, e):
```

Comment:
Is there a reason for this to be a public method? Can't see how/why an external user might call it.

If not, we probably don't need a method at all, seeing how it's only two lines and called from one place only.